### PR TITLE
[feat] Introduce stickyHeaderConfig (offset, backdrop, native driver, hide cell) and onChangeStickyIndex

### PR DESCRIPTION
**Pull request by alexrass**
### What problem does this PR solve?
Sticky headers lacked key capabilities needed in real apps:
- Could not offset headers to account for top bars (e.g., nav bars).
- No way to render a background/backdrop beneath sticky headers.
- Duplicate visuals when the original cell remained visible while its header was sticky.
- No callback to know which header is currently sticky.

This PR adds configuration and APIs to control sticky header behavior and appearance, while keeping defaults backward compatible.

### Changes
- Public API
  - Added stickyHeaderConfig with:
    - offset: top offset where headers stick (default 0).
    - backdropComponent: renders behind sticky headers.
    - useNativeDriver: controls animation driver (default true).
    - hideRelatedCell: hides the cell for the active sticky header (default false).
  - Added onChangeStickyIndex(current, previous) to track sticky header changes.
- Behavior/UX
  - Sticky header position now supports a top offset (for in-app headers).
  - Optional backdrop is layered above content but behind the sticky header.
  - Optionally hide the related cell to avoid duplicate visuals.
  - Improved push animation; when offset > 0, header fades as it’s pushed.
- Internals
  - RecyclerView:
    - Passes sticky header config to StickyHeaders.
    - Tracks currentStickyIndex to drive hideRelatedCell.
    - Applies marginTop spacer to content for offset alignment.
    - Uses configured useNativeDriver for scroll animations.
    - Renders backdrop overlay when configured.
  - StickyHeaders:
    - Computes current/next sticky indices respecting offset and content layouts.
    - Adjusts push thresholds by offset; sets absolute top = offset; higher z-index.
    - Emits onChangeStickyIndex on index transitions.
  - ViewHolder / ViewHolderCollection:
    - New hidden prop to toggle opacity when hideRelatedCell is enabled.
    - Memoization updated to include hidden.
  - useSecondaryProps:
    - New renderStickyHeaderBackdrop helper.
- Documentation and Examples
  - Docs: Added stickyHeaderConfig and onChangeStickyIndex sections with examples.
  - Fixture app: New StickyHeaderExample screen with toggles (enable sticky, offset, background).
  - Navigation updated to include the new example.
- Tests
  - New src/__tests__/StickyHeaders.test.tsx covering:
    - Sticky index selection with/without offset.
    - Push animation thresholds (including firstItemOffset).
    - Callback behavior across scrolls and edge cases.
    - Rendering position with offset.
- Changelog
  - Added entries for sticky header offset and backgrounds.

### Test Plan
- Run unit tests:
  - yarn test
  - Confirm StickyHeaders.test.tsx passes.
- Manual verification in fixture app:
  - Launch the RN fixture.
  - Open “Sticky Headers”.
  - Toggle:
    - Enable Sticky Headers: headers stick at scrolling.
    - Sticky Header Offset: headers stick below the top (e.g., 44).
    - Sticky Header Background: backdrop renders behind headers.
    - Hide cell (set in code via hideRelatedCell: true): original cell hides when its header is sticky.
- API validation:
  - Provide stickyHeaderConfig in an app using FlashList; verify default behavior remains unchanged when config is omitted.
  - Add onChangeStickyIndex and log current/previous indices; scroll through list and confirm correct sequence.

### CODE REVIEW

File: src/recyclerview/components/StickyHeaders.tsx [hunk 1]
```diff
@@ -147,28 +160,44 @@ export const StickyHeaders = <TItem,>({
...
-      extrapolate: "clamp",
-    });
-  }, [recyclerViewManager, currentStickyIndex, scrollY, pushStartsAt]);
+    return {
+      translateY: scrollY.interpolate({
+        inputRange: [pushStartsAt, pushStartsAt + currentStickyHeight],
+        outputRange: [0, -currentStickyHeight],
+        extrapolate: "clamp",
+      }),
+      opacity: stickyHeaderOffset
```
Comment (naqvitalha) [inline src/recyclerview/components/StickyHeaders.tsx]: Let's put an explicit check > 0

File: src/recyclerview/components/StickyHeaders.tsx [hunk 2]
```diff
@@ -202,6 +240,14 @@ function findCurrentStickyIndex(
   adjustedValue: number,
   getY: (index: number) => number
 ): number {
+  // Special case: when at the top of the list, show the first sticky header if it's near the top
+  if (adjustedValue <= 0 && sortedIndices.length > 0) {
```
Comment (naqvitalha) [inline src/recyclerview/components/StickyHeaders.tsx]: Can you explain this a bit more?

**Comment by benkingcode** [comment]
@alexrass Is there a way to get the backdropComponent to appear under the scrollbar? It doesn't really work for adding a background color or blur because it also affects the scrollbar

**Comment by lukebrandonfarrell** [comment]
I was just looking for this, thank you! Legend ❤️

**Comment by ayush-nutana** [comment]
can someone share videos of how it is visually looking
